### PR TITLE
update go version

### PR DESCRIPTION
**Pull request by rinckm**
### Description

Update go version to 1.16.15
Fix timeout pattern

### Submitter checklist

- [ ] Change has been tested (on a back-end cluster)
- [ ] (If applicable) Jira backlog item ID added to the PR title and the [changelog.yaml] entry
- [ ] [changelog.yaml] entry with upgrade notes is prepared and appropriate for the audience affected by the change (users or developer, depending on the change)
- [ ] Semantic version diffed against [last release][releases] and updated accordingly. In this project the version has to be maintained here:
    - [/charts/steward/Chart.yaml](https://github.com/SAP/stewardci-core/blob/master/charts/steward/Chart.yaml) (`version` and `appVersion`)

In case dependencies have been updated:
- [ ] Links to external changelogs, since the last release of our component, added to the [changelog.yaml] entry (description).
- [ ] Changelogs read thoroughly, potential impact described, upgrade notes prepared (if necessary)
- [ ] Check if dependency updates affect our semantic version increment.

### Reviewer checklist

Before the changes are marked as `ready-for-merge`: 

- [ ] There is at least 1 approval for the pull request and no outstanding requests for change
- [ ] All voter checks have passed
- [ ] Conversations in the pull request are over OR it is explicit that a reviewer does not block the change
- [ ] The Pull Request title is understandable and reflects the changes well
- [ ] The Pull Request description is understandable and well documented
- [ ] [changelog.yaml] entry for this Pull Request has been added
    - [ ] Changelog entry contains all required information
    - [ ] 'Upgrade notes' are documented in changelog.yaml (if required)

[changelog.yaml]: https://github.com/SAP/stewardci-core/changelog.yaml
[releases]: https://github.com/SAP/stewardci-core/releases

### CODE REVIEW

File: changelog.yaml [hunk 1]
```diff
@@ -59,6 +59,13 @@
       pullRequestNumber: 316
       jiraIssueNumber: CLOUDCI-1286
 
+    - type: security
+      impact: patch
+      title: Update go version
+      description: |-
+        Update go version to 1.16.15
+      pullRequestNumber: 316
```
Comment (rinckm) [inline changelog.yaml:67]:
```suggestion
      pullRequestNumber: 317
```

**Review by pavanmn22 — APPROVED** [review]

File: pkg/runctl/cfg/config_test.go [hunk 2]
```diff
@@ -243,8 +243,6 @@ func Test_withRecoverablility(t *testing.T) {
 		{"retry_on_infra_error", true, true, true},
 	} {
 		t.Run(tc.name, func(t *testing.T) {
-			tc := tc // capture current value before going parallel
-			t.Parallel()
```
Comment (LenaHoinkis) [inline pkg/runctl/cfg/config_test.go:247]: Why are those line removed? Does executing those 4 testcases in parallel not work?

File: charts/steward/crds/pipelineruns.yaml [hunk 3]
```diff
@@ -80,7 +80,7 @@ spec:
                 default: run
               "timeout": ###
                 type: string
-                pattern: '([0-9]+h)?([0-9]+m)?([0-9]+s)?'
+                pattern: '^([0-9]+h)?([0-9]+m)?([0-9]+s)?$'
```
Comment (LenaHoinkis) [inline charts/steward/crds/pipelineruns.yaml:83]: looks correct. However I would expect also a note in the changelog. If you faced some issues because of this pattern, it would be also useful to mention this in the changelog/PR.

File: pkg/runctl/cfg/config_test.go (hunk 2 above)
Comment (rinckm) [reply pkg/runctl/cfg/config_test.go:247]: The test failed centrally but this was not reproduceable locally.
Locally it was also faster without parallel execution as the test is very simple and parallel execution is much overhead.

File: charts/steward/crds/pipelineruns.yaml (hunk 3 above)
Comment (rinckm) [reply charts/steward/crds/pipelineruns.yaml:83]: The pattern was introduced newly with the prior pull which is not yet released.

**Review by LenaHoinkis — APPROVED** [review]
lgtm
